astropy/coordinates/transformations.py

```diff
@@ -1102,12 +1103,27 @@ def _apply_transform(self, fromcoord, matrix, offset):
         elif rad_vel_diff:
             data = data.without_differentials()
 
-        # Convert the representation and differentials to cartesian without
-        # having them attached to a frame
-        rep = data.to_cartesian()
-        diffs = dict([(k, diff.represent_as(CartesianDifferential, data))
-                      for k, diff in data.differentials.items()])
-        rep = rep.with_differentials(diffs)
+        # There is the special case of a rotation matrix applied on to a
```

Comment:
My sense is that this is the wrong place to tackle it. Rather, for each transformation we know whether or not it is a rotation matrix, and we can thus pass that information on (if we cannot do something simple like a flag for backwards compatibility, perhaps use a specific class or so).

Reply:
I've started on this. The BaseAffineTransformation ``_apply_transform`` now takes an argument ``is_rotation`` that is passed to the transform. I now have to propagate this up the whole transform machinery.